### PR TITLE
feat: adding filters to span details

### PR DESCRIPTION
**Pull request by palbizu**
## Description
Adding filters to span details

### Testing
Locally Tested

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: projects/observability/src/shared/components/span-detail/call/headers/span-detail-call-headers.component.ts [hunk 1]
```diff
@@ -11,13 +12,21 @@ import { EMPTY, Observable, of } from 'rxjs';
...
-      <div class="container" data-sensitive-pii>
+      <div class="container">
         <ng-container *htLoadAsync="this.records$ as records">
-          <ht-list-view
-            [records]="records"
-            [metadata]="this.metadata"
-            display="${ListViewDisplay.Plain}"
-          ></ht-list-view>
+          <ht-list-view [records]="records" [metadata]="metadata" display="${ListViewDisplay.Plain}" data-sensitive-pii>
+            <div *ngIf="!this.metadata">
```
Comment (palbizu) [inline projects/observability/src/shared/components/span-detail/call/headers/span-detail-call-headers.component.ts:18]: added to support all cases

File: projects/observability/src/shared/components/span-detail/call/headers/span-detail-call-headers.component.ts [hunk 2]
```diff
@@ -33,14 +41,24 @@ export class SpanDetailCallHeadersComponent implements OnChanges {
...
 
+  public constructor(private readonly explorerService: ExplorerService) {}
+
   public ngOnChanges(): void {
     this.buildRecords();
   }
 
+  public getExploreNavigationParams = (record: ListViewRecord): Observable<NavigationParams> =>
+    this.explorerService.buildNavParamsWithFilters(ScopeQueryParam.EndpointTraces, [
+      { field: this.filterName, subpath: record.key, operator: FilterOperator.Equals, value: record.value }
```
Comment (anandtiwary) [inline projects/observability/src/shared/components/span-detail/call/headers/span-detail-call-headers.component.ts:59]: `filterName` -> What is it representing exactly? attribute name?

File: projects/observability/src/shared/components/span-detail/call/headers/span-detail-call-headers.component.ts [hunk 3]
```diff
@@ -33,14 +41,24 @@ export class SpanDetailCallHeadersComponent implements OnChanges {
...
 
   public label?: string;
 
+  public constructor(private readonly explorerService: ExplorerService) {}
+
   public ngOnChanges(): void {
     this.buildRecords();
   }
 
+  public getExploreNavigationParams = (record: ListViewRecord): Observable<NavigationParams> =>
```
Comment (anandtiwary) [inline projects/observability/src/shared/components/span-detail/call/headers/span-detail-call-headers.component.ts:57]: nit: build<>

File: projects/observability/src/shared/components/span-detail/call/headers/span-detail-call-headers.component.ts (hunk 2 above)
Comment (palbizu) [reply projects/observability/src/shared/components/span-detail/call/headers/span-detail-call-headers.component.ts:59]: Yes, attribute name seems clearer?

Comment (anandtiwary) [reply projects/observability/src/shared/components/span-detail/call/headers/span-detail-call-headers.component.ts:59]: Yeah, or just `field` or `fieldName` is also better.

**Review by arjunlalb — APPROVED** [review]

File: projects/observability/src/shared/components/span-detail/call/headers/span-detail-call-headers.component.ts (hunk 2 above)
Comment (palbizu) [reply projects/observability/src/shared/components/span-detail/call/headers/span-detail-call-headers.component.ts:59]: done

**Review by anandtiwary — APPROVED** [review]

**Pull request merged**